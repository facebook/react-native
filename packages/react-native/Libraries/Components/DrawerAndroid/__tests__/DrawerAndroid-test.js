/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const ReactNativeTestTools = require('../../../Utilities/ReactNativeTestTools');
const View = require('../../View/View');
/* $FlowFixMe[cannot-resolve-module] (>=0.99.0 site=react_native_ios_fb) This
 * comment suppresses an error found when Flow v0.99 was deployed. To see the
 * error, delete this comment and run Flow. */
// $FlowFixMe[missing-platform-support]
const DrawerLayoutAndroid = require('../DrawerLayoutAndroid.android');
const React = require('react');

describe('<DrawerLayoutAndroid />', () => {
  it('should render as expected', async () => {
    await ReactNativeTestTools.expectRendersMatchingSnapshot(
      'DrawerLayoutAndroid',
      () => (
        <DrawerLayoutAndroid
          drawerWidth={300}
          drawerPosition="left"
          renderNavigationView={() => <View />}
        />
      ),
      () => {
        jest.dontMock('../DrawerLayoutAndroid');
      },
    );
  });
});
