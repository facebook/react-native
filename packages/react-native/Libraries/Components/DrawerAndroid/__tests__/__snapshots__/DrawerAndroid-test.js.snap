// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerLayoutAndroid /> should render as expected: should deep render when mocked (please verify output manually) 1`] = `
<AndroidDrawerLayout
  drawerBackgroundColor="white"
  drawerPosition="left"
  drawerWidth={300}
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  onDrawerSlide={[Function]}
  onDrawerStateChanged={[Function]}
  style={
    Array [
      Object {
        "elevation": 16,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "backgroundColor": "white",
          "width": 300,
        },
      ]
    }
  >
    <View />
  </View>
</AndroidDrawerLayout>
`;

exports[`<DrawerLayoutAndroid /> should render as expected: should deep render when not mocked (please verify output manually) 1`] = `
<AndroidDrawerLayout
  drawerBackgroundColor="white"
  drawerPosition="left"
  drawerWidth={300}
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  onDrawerSlide={[Function]}
  onDrawerStateChanged={[Function]}
  style={
    Array [
      Object {
        "elevation": 16,
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "top": 0,
        },
        Object {
          "backgroundColor": "white",
          "width": 300,
        },
      ]
    }
  >
    <View />
  </View>
</AndroidDrawerLayout>
`;
